### DISCUSSION TITLE
两个路由指向同一个Form组件，不同的路由加载组件时能否重新渲染

### DISCUSSION THREAD

**Discussion by HengzheLi** [alibaba/formily#3337] [Q&A 2.x] [answered]
https://codesandbox.io/s/bold-bas-7j5rvl?file=/navigation.tsx

路由"/create"和"/edit"都指向“App”组件，先点击"编辑"，使用"/edit"打开App，在输入框中随便输入。然后点击"创建"，再使用"/create"打开App组件，但此时组件中的输入框保留有/edit时编辑的数据。

React新人，不确定这个是react的特性还是formily的bug。希望在/edit编辑后再切换到/create时，App能重新渲染组件，但是现在看应该是使用了上一次的渲染结构

**Comment by bjlsy** [comment]
组件加key

**Comment by janryWang** [comment, marked as answer]
form实例化放组件里，useMemo(()=>createForm(),[])
